config.go

```diff
@@ -535,7 +535,17 @@ func loadConfig() (*config, []string, error) {
 	// All data is specific to a network, so namespacing the data directory
 	// means each individual piece of serialized data does not have to
 	// worry about changing names per network and such.
+	//
+	// Check for old versions of testnet directories here since the network
+	// specific DataDir will be used after this.
 	cfg.DataDir = cleanAndExpandPath(cfg.DataDir)
+	var oldTestNet bool
+	oldDataDir := filepath.Join(cfg.DataDir, "testnet")
+	if cfg.TestNet {
```

Comment:
I'm ambivalent.  I sort of like it the way it is where it only warns when running with `--testnet`.  I'm not sure I care when running on `--mainnet` that a testnet that I once used months ago is no longer recent.